com.gitlab.coringao.cavestory-nx.json

```diff
@@ -0,0 +1,42 @@
+{
+  "app-id": "com.gitlab.coringao.cavestory-nx",
+  "runtime": "org.freedesktop.Platform",
+  "runtime-version": "18.08",
+  "sdk": "org.freedesktop.Sdk",
+  "command": "cavestory-nx",
+  "finish-args": [
+    "--device=all",
+    "--socket=pulseaudio",
+    "--socket=x11"
```

Comment:
You should probably enable faster X11 display, using XShm extension:
```
  "--share=ipc"
```

And for Wayland, you need this:
```
  "--socket=wayland"
```

Reply:
Have you tested it on wayland?

Reply:
Not yet.

Reply:
@cpba 
By default, it uses XWayland. You can force native Wayland by setting `SDL_VIDEODRIVER=wayland`, e.g:
```
flatpak-builder --run build com.gitlab.coringao.cavestory-nx.json sh -c 'SDL_VIDEODRIVER=wayland cavestory-nx'
```
Of course, you have to remember to put the `--socket=wayland` in the manifest.

However, it doesn't work very well, especially when you changing resolution and switching between window mode and fullscreen. Moreover, there is no window border in window mode, so it's not a pleasant experience.